modules/terraform/options.go

```diff
@@ -18,4 +18,5 @@ type Options struct {
 	Upgrade                  bool                   // Whether the -upgrade flag of the terraform init command should be set to true or not
 	NoColor                  bool                   // Whether the -no-color flag will be set for any Terraform command or not
 	SshAgent                 *ssh.SshAgent          // Overrides local SSH agent with the given in-process agent
+	Lock                     bool                   // Enable locking for plan, apply and destroy (by default only plan uses locking)
```

Comment:
Oh! I'm not sure why that's there actually. We should not disable locking arbitrarily...

Reply:
In the plan command, it is currently enabled as well:
https://github.com/gruntwork-io/terratest/blob/master/modules/terraform/plan.go#L36

Would it be worth having TerraformOptions, or inidividual options in the Options struct so that any of the terraform options can be set or overridden?

Reply:
> In the plan command, it is currently enabled as well: https://github.com/gruntwork-io/terratest/blob/master/modules/terraform/plan.go#L36

Doesn't seem quite right...

> Would it be worth having TerraformOptions, or inidividual options in the Options struct so that any of the terraform options can be set or overridden?

Not sure I understand the question?

Reply:
Sorry, I wasn't very clear.
I was suggesting an extra field in the Options struct (in a similar way to the Vars field [here](https://github.com/gruntwork-io/terratest/blob/361f35fd139bcd05918f72fcea42dff0060c776f/modules/terraform/options.go#L12)) that could hold the options that need to be passed to the terraform command.

For example:
`type Options struct {
	TerraformDir             string                 // The path to the folder where the Terraform code is defined.
        Vars                     map[string]interface{} // The vars to pass to Terraform commands using the -var option.
	...
        TerraformOptions // The vars to pass to the Terraform commands (lock, lock-timeout etc.)
         `

Reply:
It's probably worth exposing `Lock` as a top-level item in `terraform.Options`. However, adding an `ExtraArguments []string` field to let users pass arbitrary other arguments to terraform commands is a good idea too.

Reply:
Also, I think disabling locking by default for all commands is fine.